File: binary_search.py
def search(array,item):
    return __search(array,item,0,len(array))

def __search(array,item,start,end):
    if end-start>1:
        middleIndex=int((start+end)/2)
        middle=array[middleIndex]
        if middle<item:
            return __search(array,item,middleIndex+1,len(array))
        elif middle>item:
            return __search(array,item,0,middleIndex)
        else:
             return middleIndex
    else:
        if array[start]==item:
            return start
        else:
            return -1

File: src/app/binarysearch/BinarySearch.java
package app.binarysearch;

public class BinarySearch
{
    public int search(int[] array,int element)
    {
        return search(array, element, 0, array.length);
    }
    private int search(int[] array,int element,int start,int end)
    {
        if(end-start==1)
        {
            if(array[start]==element)
                return start;
            return -1;
        }
        int middleIndex=(start+end)/2;
        if(array[middleIndex]>element)
            return search(array, element,0,middleIndex);
        else if(array[middleIndex]>element)
            return search(array, element,middleIndex,end);
        return middleIndex;
    }
}